### ISSUE TITLE
Add timeout on mod operations

### ISSUE THREAD

**Issue by bluepichu** [bluepichu/a-la-mod#53]
Though how long it should be is up for debate.

**Comment by bluepichu** [comment]
Or maybe that should even be a user-specific setting?

**Comment by DelrithInfinity** [comment]
I think it would work well a user specific setting: this could even mean that when you include a new module and it sees your low timeout, it could ask you to turn it higher.

**Comment by bluepichu** [comment]
Further, should it be *mod*-specific?  That could make sense as well - something to parse links shouldn't take more than a few tenths of a second at most, but something doing a heavy image process should be allocated much more time.

**Comment by DelrithInfinity** [comment]
Yes, it should be. There should also be group settings. I group modules as in text modifiers for .1 seconds because I know they should run fast. Also: a default group.
